### PR TITLE
Increase and decrease capacity Test/Helm chart valid Test/Are any containers running in privileged mode Test

### PR DESCRIPTION
**Pull request by wavell**
## Description
Scalability Test: Can CNF increase and decrease capacity?
Installability Test: Helm chart valid? 
Security test: Are any containers running in privileged mode?
Issues:
- https://github.com/cncf/cnf-conformance/issues/24
- https://github.com/cncf/cnf-conformance/issues/6
- https://github.com/cncf/cnf-conformance/issues/55

## How Has This Been Tested?
* [ ]  Covered by existing integration testing
* [ ]  Added integration testing to cover
* [ ]  Tested manually
   * [ ] tag
   * [ ] master
   * [x] develop
* [x]  Tested locally
* [ ]  Have not tested

## Types of changes
* [ ]  Bug fix (non-breaking change which fixes an issue)
* [x]  New feature (non-breaking change which adds functionality)
* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
 
## Checklist:
* [ ]  My change requires a change to the documentation.
* [x]  I have updated the documentation accordingly.
* [ ] No updates required.

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 ---
 helm_directory: cnfs/cnf-testbed/examples/use_case/gw-routers-on-k8s-nsm-on-packet/helm/gwr 
 install_script: cnfs/cnf-testbed/examples/network_functions/rdma_vppcontainer/builder.sh
-container_name: priv2-coredns 
+deployment_name: coredns-coredns
+application_deployment_names: [coredns-coredns]
+helm_chart: stable/coredns
```
Comment (taylor) [inline config.yml]: How does this correspond to a path for the chart?  

@wavell @denverwilliams

File: USAGE.md [hunk 2]
```diff
@@ -80,9 +80,21 @@ crystal src/cnf-conformance.cr protected_access
 crystal src/cnf-conformance.cr scaling
 ```
 
-#### (WIP) To test the [increasing and decreasing of capacity](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#scaling-resources)
+#### :heavy_check_mark To test the [increasing and decreasing of capacity](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#scaling-resources)
+Optional: To install the sample coredns cnf:
+
+```
+crystal src/cnf-conformance.cr sample_coredns_setup helm_chart=<helm chart name>
```
Comment (taylor) [inline USAGE.md]: @wavell @wavell does this mean it automatically downloads the chart named?

File: sample-cnfs/sample-coredns-cnf/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```
+helmenv versions 
+helmenv install <version 3.1?>
+```
+
+### core-dns installation
+```
+helm install coredns stable/coredns
+```
+### Pull down the helm chart code, untar it, and put it in the cnfs/coredns directory
```
Comment (taylor) [inline sample-cnfs/sample-coredns-cnf/README.md]: @wavell @denverwilliams I'm not sure how this instruction for downloading the chart with helm by name works with the config file which specifies a name but no path.

File: src/cnf_conformance.cr [hunk 4]
```diff
@@ -1,4 +1,3 @@
-require "tasks/declarative/*"
 # TODO: Write documentation for `CnfConformance`
 module CnfConformance
   VERSION = "0.1.0"
```
Comment (taylor) [inline src/cnf_conformance.cr]: @wavell when do we bump the version?

**Comment by taylor** [comment]
@denverwilliams @wavell @williscool @nupejosh

File: src/tasks/utils.cr [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # TODO make constants local or always retrieve from environment variables
```
Comment (williscool) [inline src/tasks/utils.cr]: I would get rid of this in favor of using commander for it in the rest of the code base too

File: config.yml (hunk 1 above)
Comment (wavell) [reply config.yml]: The path for the chart is required for the helm lint ticket.  Not this one. @taylor

File: USAGE.md (hunk 2 above)
Comment (wavell) [reply USAGE.md]: Yes

File: sample-cnfs/sample-coredns-cnf/README.md (hunk 3 above)
Comment (wavell) [reply sample-cnfs/sample-coredns-cnf/README.md]: You don't need a path for the helm chart download for this ticket

File: src/cnf_conformance.cr (hunk 4 above)
Comment (wavell) [reply src/cnf_conformance.cr]: maybe when it is tagged

File: sample-cnfs/sample-coredns-cnf/cnf-conformance.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+helm_directory: cnfs/coredns/helm_chart/coredns
+install_script: cnfs/coredns/Makefile
+deployment_name: coredns-coredns 
+application_deployment_names: [coredns-coredns]
+helm_chart: stable/coredns
+helm_chart_container_name: coredns
+white_list_helm_chart_container_names: [falco, nginx, coredns, calico-node, kube-proxy, nginx-proxy]
```
Comment (taylor) [inline sample-cnfs/sample-coredns-cnf/cnf-conformance.yml]: I do not understand _white_list_helm_chart_container_names_  having chart in the name.  What is this key for?

also this key says containers but the test is actually checking pods.

It seems like the name should either match what it's checking or be more generic. Example:
- white_list_pod_names
- white_list_cnf_names

cc @wavell @denverwilliams

File: src/tasks/configuration_lifecycle.cr [hunk 7]
```diff
@@ -10,56 +10,68 @@ end
...
+      end
     end
+    if response.to_s.size > 0
+      puts "FAILURE: IP addresses found".colorize(:red)
+    else
+      puts "PASSED: No IP addresses found".colorize(:green)
+    end
+    Dir.cd(cdir)
+  rescue ex
+    puts ex.message
```
Comment (taylor) [inline src/tasks/configuration_lifecycle.cr]: @wavell @denverwilliams this entire section seems unrelated to this PR.
